packages/dapp/src/pages/Search.tsx

```diff
@@ -145,9 +145,9 @@ export const Search = () => {
         </Box>
       </MessageBox>
 
-      {loading ? <Spinner color='accent-1' alignSelf='center' size='large' /> : null}
+      {loading || afterLoading ? <Spinner color='accent-1' alignSelf='center' size='large' /> : null}
 
-      <MessageBox type='info' show={!afterLoading && noResults}>
```

Comment:
noRsults is the only way to know if results exist or not right after the search.
state update takes time and until this state is not updated you cannot know if the search is succeeded or not.